docs/building-spark.md

```diff
@@ -79,15 +81,20 @@ Example:
 
     ./build/mvn -Pyarn -Dhadoop.version=2.8.5 -DskipTests clean package
 
+    ./build/mvn -Pyarn -Phadoop-3.2 -Dhadoop.version=3.1.2 -DskipTests clean package
```

Comment:
Also, could you update the PR title because this PR includes this `hadoop` example update additionally?

Reply:
just for curiosity, why we can use `-Phadoop-3.2` with hapdoop version 3.1.2?

Reply:
Specify the built-in Hive to 2.3 by `-Phadoop-3.2`.

Reply:
I think Wenchen asked why we set `hadoop-3.2` (Hadoop 3.2) but hadoop version is  Hadoop 3.1.2. I just noticed it. Is it something we can officially document? It might work practically but not sure if it's something official.

I think Spark documents `hadoop2.7` supports Hadoop 2.7+ assuming the versions afterward keeps backward compatibility. But I am not sure if that's true in the opposite way.

Reply:
I would just remove this line. The one above already illustrates a customized Hadoop version.